zerver/lib/actions.py

```diff
@@ -6358,31 +6363,16 @@ def do_update_message_flags(
     flagattr = getattr(UserMessage.flags, flag)
 
     msgs = UserMessage.objects.filter(user_profile=user_profile, message_id__in=messages)
-    # This next block allows you to star any message, even those you
-    # didn't receive (e.g. because you're looking at a public stream
-    # you're not subscribed to, etc.).  The problem is that starring
-    # is a flag boolean on UserMessage, and UserMessage rows are
-    # normally created only when you receive a message to support
-    # searching your personal history.  So we need to create one.  We
-    # add UserMessage.flags.historical, so that features that need
-    # "messages you actually received" can exclude these UserMessages.
-    if msgs.count() == 0:
-        if not len(messages) == 1:
-            raise JsonableError(_("Invalid message(s)"))
-        if flag != "starred":
-            raise JsonableError(_("Invalid message(s)"))
-        # Validate that the user could have read the relevant message
-        message = access_message(user_profile, messages[0])[0]
```

Comment:
It is essential that we call `access_message` in this code path; the reactions code path already does it at the beginning, but this one does not, and so merging this change would introduce a serious security bug.

Reply:
Oh, you in fact do this; I just didn't quite read it correctly.  Great.

Reply:
OK and I also confirmed that tests fail if we remove the `access_message` check in the revised version, so we're good here.